tests/Common/Client/S3ClientFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Common\Client;
+
+use Aws\MockHandler;
+use Aws\S3\S3Client;
+use Aws\S3\S3ClientInterface;
+
+class S3ClientFactory
+{
+    public static function create(MockHandler $handler): S3ClientInterface
+    {
+        $s3client = new S3Client([
+            'version'     => $_ENV['AWS_S3_VERSION'],
+            'region'      => $_ENV['AWS_S3_REGION'],
+            'credentials' => [
+                'key'    => '',
+                'secret' => '',
+            ],
+            'handler' => $handler,
+        ]);
+
+        return $s3client;
+    }
+}
```

Comment:
```suggestion
use App\Aws\S3Config;
use Aws\MockHandler;
use Aws\S3\S3Client;
use Aws\S3\S3ClientInterface;

class S3ClientFactory
{
    public static function create(MockHandler $handler, S3Config $s3Config): S3ClientInterface
    {
        $s3client = new S3Client([
            'version'     => $s3Config->version,
            'region'      => $s3Config->region,
            'credentials' => [
                'key'    => '',
                'secret' => '',
            ],
            'handler' => $handler,
        ]);

        return $s3client;
    }
}
```
Avoid interacting with a global state.